reports/models.py

```diff
@@ -0,0 +1,459 @@
+import logging
+from datetime import datetime
+from typing import List, Optional, Union
+
+import occurrences.models as occurrences_models
+import reports.services as reports_services
+from django.contrib.postgres.fields import ArrayField
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from reports.exceptions import EnrolmentReportCouldNotHydrateLinkedEventsData
+from reports.utils import haversine
+
+from common.models import TimestampedModel
+
+logger = logging.getLogger(__name__)
+
+
+class EnrolmentQuerySet(models.QuerySet):
```

Comment:
I find it a bit confusing that we have a a manager/queryset on a model that returns a completely different model instances than what the `self.model` in this case would be. Also there's maybe some hacky way of using self here.

Maybe we could instead just have this functionality as functions? 

```python
def get_unsynched_enrollments(
    sync_from: datetime = None,
) -> Union[models.QuerySet, List[occurrences_models.Enrolment]]:
    """Get a list of enrolments which are updated after latest report updates."""
    sync_from = sync_from or EnrolmentReport.objects.latest_sync()
    if sync_from:
        return occurrences_models.Enrolment.objects.filter(updated_at__gte=sync_from)
    return occurrences_models.Enrolment.objects.all()


def get_missing_enrollments(
    sync_from: datetime = None,
) -> Union[models.QuerySet, List[occurrences_models.Enrolment]]:
    """Get a list of enrolments which are missing from EnrolmentReport db-table."""
    enrolments = get_unsynched_enrollments(sync_from=sync_from)
    reports_enrolment_ids = EnrolmentReport.objects.filter(
        _enrolment_id__in=[e.id for e in enrolments]
    ).values_list("_enrolment_id", flat=True)
    return enrolments.exclude(id__in=reports_enrolment_ids)
```